### ISSUE TITLE
Feature Request: Retrieve data with custom extent

### ISSUE THREAD

**Issue by NilsBrehm** [G-Node/nixpy#298]
Hey guys,
I think it would be handy if you could retrieve data of a tag (or multi tag) from position to a custom extent.
Sometimes the interval between position and extent is too short (e.g. when using short stimuli). With some help from Jan I managed to do use using some extra code. But it would be nice to have this as a included function.
Best
Nils

**Comment by achilleas-k** [comment]
Seems like it wouldn't be too much trouble to implement and shouldn't interfere with the current state.

Let me have a look next week and I'll link you to a branch you can try out. Sound good?

**Comment by achilleas-k** [comment]
So this is extremely straightforward to implement, even though I neglected it for a while. But now I'm thinking, how do we handle the C++ HDF5 backend? Should we add this now and have it be not supported with the HDF5 backend, or do we wait until we're decoupled for the bindings and call it a feature of the Python API?

@jgrewe & @gicmo Thoughts?

**Comment by gicmo** [comment]
I would say, go ahead, and forget about the C++ side of things for now.

**Comment by jgrewe** [comment]
I was wondering if a more generic method that returns the data from a DataArray using the dimensions would be a better solution than extending the retrieveData method. It could live on the DataArray or the Tag/MTag and simply take starts and ends in dimension values. For example something like this:

``` 
da = b.create_data_array("test", "test", data=2d_data)
da.append_sampled_dimension(0.1)
da.append_sampled_dimension(0.01)

da.get_data((0.0, 0.0), (10.0, 0.1))  # in units of the dimension(s), not indices
```
This would be more flexible. Fixing the existing retrieve_data would probably lead to the next situation, in which a negative offset, i.e. data before (not only after) the position is requested.

**Comment by achilleas-k** [comment]
Yeah, that does sound nice. Basically a dimension-based slicing method as a natural alternative to the indexed slicing. Definitely more useful for quick lookups.

What would we call it? `get_data` isn't wrong, but it's not descriptive either. I'd like to call it something that makes it clear the user is slicing based on the dimensions. Another option might be to have a generic slicing method and have it take an argument that makes it retrieve data based on index or dim units. Possibly uglier though.

**Comment by achilleas-k** [comment]
Enum is a good idea. `get_slice` is up for debate, as a name.

New question: Should the return value be a DataView or a raw numpy array?

**Comment by jgrewe** [comment]
good question. I would prefer to stay consistent with the other methods. Since the retrieve data functions return a DataView, I would vote for that

**Comment by achilleas-k** [comment]
I figured that would be best, yeah.
